### ISSUE TITLE
Static code scan (Coverity) issue seen in file: /subsys/net/lib/sockets/sockets.c

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#2366]
**_Reported by Shilpashree L C:_**

Coverity issue seen with CID: 173646 
 Category: Error handling issues 
 Function: zsock_close
 Please fix or provide comments to square it off in coverity : https://scan9.coverity.com/reports.htm#v31655/p12996

(Imported from Jira ZEP-2536)

**Comment by nashif** [comment]
**_by Paul Sokolovsky:_**

Looking at that case, there's a code comment right above it (added by _Jukka Rissanen_ ) telling:

{quote}No need to check return values here
* as Coverity complains in CID 173646.{quote}

As _Jukka Rissanen_  mentioned previously, he or me can't change anything in Coverity UI directly to mark it a False Positive.

What I can also propose is to explicitly cast function return value to void, e.g.

{code:java}
(void)net_context_accept(ctx, NULL, K_NO_WAIT, NULL);
```

I'm not sure if Coverity will treat it as expected. Please suggest.

**Comment by nashif** [comment]
**_by Shilpashree L C:_**

_Paul Sokolovsky_  Can you confirm if you have access to the Coverity? If so please provide me with your username so that I can provide specific access rights to do the changes, or else please request for access in Coverity website.

**Comment by nashif** [comment]
**_by Paul Sokolovsky:_**

_Shilpashree L C_ , yes, I can login, can search for CID, see the code in the left pane. However, in the right pane, titled "Triage", after change some fields, both "Apply + Next" and "Apply" buttons remain grayed-out, so I can't save changes.

I login to Coverity via Github authentication, so my account name in the top right corner appears as "pfalcon@users.sourceforge.net".

**Comment by nashif** [comment]
**_by Shilpashree L C:_**

_Paul Sokolovsky_  For your username the User role was Defect Viewer (Read only), so you were not able to do any changes. I have now changed your role as Contributor/Member, I think now you will be able to make changes.

**Comment by nashif** [comment]
**_by Shilpashree L C:_**

Duplicate of GH-2296

**Comment by nashif** [comment]
**_by Paul Sokolovsky:_**

_Shilpashree L C_ : Sorry, I went on vacation and missed to reply to this ticket earlier. Thanks for giving permission to triage Coverity issues, I confirm that it works. And while this was ignore in Coverity, I did a bit more homework how to avoid such cases in the future, and submitted https://github.com/zephyrproject-rtos/zephyr/pull/1327 . In either case, this issue is resolved, not further actions needed, I'm posting this just FYI. Thanks!
